platform/mellanox/mlnx-platform-api/sonic_platform/device_data.py

```diff
@@ -256,6 +264,18 @@ def get_cpld_component_list(cls):
             return ComponentCPLDSN2201.get_component_list()
         return ComponentCPLD.get_component_list()
 
+    @classmethod
+    def get_platform_json_data(cls):
```

Comment:
We don't need the whole platform.json data. Only the DPU part.
@classmethod
@utils.read_only_cache()
def get_dpus_data(cls):
    from sonic_py_common import device_info
    platform_path = device_info.get_path_to_platform_dir()
    platform_json_path = os.path.join(platform_path, 'platform.json')
    platform_json = utils.load_json_file(platform_json_path)
    return platform_json['DPUS']

Reply:
Using try/except to return 0 by default in case of exception (2 causes of exception could be existences of `DPUS` key in platform_json data, or file could not be read)